src/openforms/js/components/admin/form_design/registrations/objectsapi/V2ConfigFields.js

```diff
@@ -39,12 +41,31 @@ const V2ConfigFields = ({index, name, schema, formData, onChange}) => {
 
   useEffect(() => {
     if (schema.properties.objectsApiGroup.enum.length === 1 && objectsApiGroup === '') {
-      onChange({
+      onFieldChange({
         target: {name: 'objectsApiGroup', value: schema.properties.objectsApiGroup.enum[0]},
       });
     }
   }, []);
 
+  const clearMappingOnChange = message => {
+    return event => {
+      if (variablesMapping.length === 0) {
+        onFieldChange(event);
+      } else {
+        const confirmSwitch = window.confirm(message);
+        if (confirmSwitch) {
+          const {name, value} = event.target;
+          onChange(
+            produce(formData, draft => {
+              draft.variablesMapping = [];
+              draft[name] = value;
+            })
```

Comment:
Using chained `onChange` (now `onFieldChange`) calls would result in race conditions :/